src/unitxt/metrics.py

```diff
@@ -151,30 +207,10 @@ def metric(sample_refs, sample_preds, sample_additional_inputs):
                 arr=arr,
             )
 
-            # when running with bca interval (default), the statistic is called twice: with the
-            # original data and with the resamples. here we want to focus only on the latter.
-            if scores.size > 1:
-                # here we deal with samples on which the metric could not be computed. These are
```

Comment:
Maybe maintain this comment in `resample_from_non_nan`? WDYT? it explains the rationale nicely ..

Reply:
I can leave the original comment, you are right.  I don't think we need the conditioning on scores.size > 1 though.

Reply:
Do you know why is was there in the first place?
I see this comment: 

# when running with bca interval (default), the statistic is called twice: with the
# original data and with the resamples. here we want to focus only on the latter.

seems it is to prevent NaN removal when running on the original data (without resamples). But not sure i understand why. Do you understand this part?